platforms/enterprise/enterprise-operations/src/main/java/org/gradle/internal/configurationcache/ConfigurationCacheCheckFingerprintBuildOperationType.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.configurationcache;
+
+import org.gradle.internal.operations.BuildOperationType;
+
+import java.util.List;
+
+/**
+ * Details about the configuration cache fingerprint check operation.
+ *
+ * @since 8.10
+ */
+public final class ConfigurationCacheCheckFingerprintBuildOperationType implements BuildOperationType<ConfigurationCacheCheckFingerprintBuildOperationType.Details, ConfigurationCacheCheckFingerprintBuildOperationType.Result> {
```

Comment:
🤔 What is the relation of this operation with `ConfigurationCacheLoadBuildOperationType` and `ConfigurationCacheStoreBuildOperationType`? It seems to be somewhat different, especially when not running tasks.

Reply:
> 🤔 What is the relation of this operation with `ConfigurationCacheLoadBuildOperationType` and `ConfigurationCacheStoreBuildOperationType`? It seems to be somewhat different, especially when not running tasks.

`VALID` check is likely to be complemented by `Load` operation, as it represents a cache hit. Other statuses may be complemented by `Store + Load` (when running tasks with CC) or just `Store` (when loading models with IP). Neither of the operations is nested into another, and I don't want to promise a specific order, though the current implementation emits fingerprint check first.